### PR TITLE
Improve existing e2e repeat tests a bit

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- adds a test to check changing the variable values updates correctly
- adds a test to check loading with query params works correctly

**Which issue(s) this PR fixes**:

For https://github.com/grafana/grafana/issues/42512

**Special notes for your reviewer**:

### CODE REVIEW

File: e2e/dashboards-suite/Repeating_a_panel_horizontally.spec.ts [hunk 1]
```diff
@@ -2,14 +2,49 @@ import { e2e } from '@grafana/e2e';
...
     const panelTitles = ['Panel Title 1', 'Panel Title 2', 'Panel Title 3'];
     panelTitles.forEach((title) => {
+      e2e.components.Panels.Panel.title(title).should('be.visible');
+    });
+
+    // Change to only show panels 1 + 3
+    e2e.pages.Dashboard.SubMenu.submenuItemLabels('horizontal').click();
+    e2e.pages.Dashboard.SubMenu.submenuItemValueDropDownOptionTexts('1').click();
+    e2e.pages.Dashboard.SubMenu.submenuItemValueDropDownOptionTexts('3').click();
+    e2e.pages.Dashboard.DashNav.navV2().click();
```
Comment (ashharrison90) [inline e2e/dashboards-suite/Repeating_a_panel_horizontally.spec.ts:43]: this is a hack which works but isn't particularly robust... would love a better suggestion for this bit. we need to blur the dropdown menu for the changes to be processed, but i can't find a way to blur it/anything suitable to select to force a blur. in the UI, you would click "outside" of the dropdown.

Comment (kaydelaney) [reply e2e/dashboards-suite/Repeating_a_panel_horizontally.spec.ts:43]: Maybe something like 
```suggestion
    e2e().get('body').click();
```

?

**Review by kaydelaney — APPROVED** [review]
Looks good! Left one small suggestion, but feel free to ignore it

File: e2e/dashboards-suite/Repeating_a_panel_horizontally.spec.ts (hunk 1 above)
Comment (ashharrison90) [reply e2e/dashboards-suite/Repeating_a_panel_horizontally.spec.ts]: yep, that's much nicer 😄  thanks!

**Pull request merged**